src/qibolab/instruments/rfsoc.py

```diff
@@ -1,54 +1,156 @@
+    elif isinstance(pulse.shape, Gaussian):
+        rfsoc_pulse.shape = "gaussian"
+        rfsoc_pulse.rel_sigma = pulse.shape.rel_sigma
+    elif isinstance(pulse.shape, Drag):
+        rfsoc_pulse.shape = "drag"
+        rfsoc_pulse.rel_sigma = pulse.shape.rel_sigma
+        rfsoc_pulse.beta = pulse.shape.beta
+    return rfsoc_pulse
+
+
+def convert_frequency_sweeper(sweeper: rfsoc.Sweeper, sequence: PulseSequence, qubits: Dict[int, Qubit]):
+    """Converts frequencies for `qibosoq.abstract.Sweeper` considering LO and HZ_TO_MHZ"""
+    for idx, jdx in enumerate(sweeper.indexes):
+        if sweeper.parameter[idx] is rfsoc.Parameter.frequency:
+            pulse = sequence[jdx]
+            try:
+                lo_frequency = getattr(qubits[pulse.qubit], pulse.type.name.lower()).local_oscillator._frequency
+            except NotImplementedError:
+                lo_frequency = 0
+
+            sweeper.starts[idx] = (sweeper.starts[idx] - lo_frequency) * HZ_TO_MHZ
+            sweeper.stops[idx] = (sweeper.stops[idx] - lo_frequency) * HZ_TO_MHZ
+
+
+def convert_sweep(sweeper: Sweeper, sequence: PulseSequence, qubits: List[Qubit]) -> rfsoc.Sweeper:
+    """Convert `qibolab.sweeper.Sweeper` to `qibosoq.abstract.Sweeper`"""
+
+    parameters = []
+    starts = []
+    stops = []
+    indexes = []
+
+    if sweeper.parameter is Parameter.bias:
+        for qubit in sweeper.qubits:
+            parameters.append(rfsoc.Parameter.bias)
+            for idx, seq_qubit in enumerate(qubits):
+                if qubit == seq_qubit:
+                    indexes.append(idx)
+            starts.append(sweeper.values[0] + qubits[qubit].flux.bias)
+            stops.append(sweeper.values[-1] + qubits[qubit].flux.bias)
+
+        if max(np.abs(starts)) > 1 or max(np.abs(stops)) > 1:
+            raise ValueError("Sweeper amplitude is set to reach values higher than 1")
+    else:
+        for pulse in sweeper.pulses:
+            for idx, seq_pulse in enumerate(sequence):
+                if pulse == seq_pulse:
+                    indexes.append(idx)
+
+            if sweeper.parameter is Parameter.amplitude:
+                parameters.append(rfsoc.Parameter.amplitude)
+                starts.append(sweeper.values[0] * pulse.amplitude)
+                stops.append(sweeper.values[-1] * pulse.amplitude)
+            elif sweeper.parameter is Parameter.relative_phase:
+                parameters.append(rfsoc.Parameter.relative_phase)
+                starts.append(np.degrees(sweeper.values[0] + pulse.relative_phase))
+                stops.append(np.degrees(sweeper.values[-1] + pulse.relative_phase))
+            elif sweeper.parameter is Parameter.frequency:
+                # frequency are not convertet yet and LO is not accounted for
+                parameters.append(rfsoc.Parameter.frequency)
+                starts.append(sweeper.values[0] + pulse.frequency)
+                stops.append(sweeper.values[-1] + pulse.frequency)
```

Comment:
This seems to me the same code three times, with small differences, but already properly structured.

So, an alternative, could be something like:
```suggestion
            if sweeper.parameter in {Parameter.amplitude, Parameter.relative_phase, Parameter.frequency}:
                name = sweeper.parameter.name
                parameters.append(getattr(rfsoc.Parameter, name))
                value = getattr(pulse, name)
                starts.append(sweeper.values[0] * value)
                stops.append(sweeper.values[-1] * value)
```

Reply:
Moreover, if I understood correctly, it is only done for those three because the others are somehow not implemented.
Then, in principle, you could even remove the `if`, since here there might be nothing failing, and postpone the error to the moment when something is missing (quite a Pythonic approach).

On the other side, it might even fail immediately, with a sensible error, if parameters that are not supported are not enumerated by `rfsoc.Parameter`.

Reply:
I implemented the suggestion with some changes. In particular differentiating between factors and offsets. 
Also note that rfsoc.Parameter has an option (time) that is supported by the server, but not yet by the client